src/FixedDataTableCellGroup.js

```diff
@@ -65,74 +69,155 @@ class FixedDataTableCellGroupImpl extends React.Component {
-      var recyclable = columnProps.allowCellsRecycling && !isColumnReordering;
-      if (!recyclable || (
-        currentPosition - props.left <= props.width &&
-        currentPosition - props.left + columnProps.width >= 0)) {
-        var key = columnProps.columnKey || 'cell_' + i;
-        cells[i] = this._renderCell(
-          props.rowIndex,
-          props.rowHeight,
-          columnProps,
-          cellTemplate,
-          currentPosition,
-          key,
-          contentWidth,
-          isColumnReordering
-        );
-      }
-      currentPosition += columnProps.width;
-    }
-    var style = {
+    // get list of cells
+    this._staticCellArray = this._computeVirtualizedCells({ columnGroupWidth, isColumnReordering });
+    const nonVirtualizedCells = this._computeNonVirtualizedCells({ columnGroupWidth, isColumnReordering });
+
+    const style = {
       height: props.height,
       position: 'absolute',
-      width: contentWidth,
+      width: columnGroupWidth,
       zIndex: props.zIndex,
     };
+
     FixedDataTableTranslateDOMPosition(style, -1 * DIR_SIGN * props.left, 0, this._initialRender);
 
     return (
       <div
         className={cx('fixedDataTableCellGroupLayout/cellGroup')}
-        style={style}>
-        {cells}
+        style={style}
+      >
+        {this._staticCellArray}
+        {nonVirtualizedCells}
       </div>
     );
   }
 
-  _renderCell = (
-    /*number*/ rowIndex,
-    /*number*/ height,
-    /*object*/ columnProps,
-    /*object*/ cellTemplate,
-    /*number*/ left,
-    /*string*/ key,
-    /*number*/ columnGroupWidth,
-    /*boolean*/ isColumnReordering,
-  ) /*object*/ => {
+  /**
+   * Return list of cells only present in the view port.
+   * Uses this._staticCellArray and columnsToRender to maintain buffer positions, thus
+   * minimizing cell mounts/unmounts.
+   * Offsets for the cells are calculated by the reducer.
+   */
+  _computeVirtualizedCells = ({
+    /*number*/columnGroupWidth,
+    /*boolean*/isColumnReordering,
+  }) => {
+    const { allowColumnVirtualization, columnsToRender, columnOffsets, isScrolling } = this.props;
+
+    if (!columnsToRender || !allowColumnVirtualization) {
+      return [];
+    }
+
+    const virtualizedCells = [];
+    if (isScrolling) {
+      virtualizedCells.length = Math.max(this._staticCellArray.length, columnsToRender.length);
+    } else {
+      // this is done so that only cells inside the buffer are considered for vertical scrolling
+      virtualizedCells.length = columnsToRender.length;
+    }
+
+    // render each cell
+    for (let staticIndex = 0; staticIndex < virtualizedCells.length; staticIndex++) {
+      let columnIndex = columnsToRender[staticIndex];
+
+      // if cell at this static index no longer exists, then just update the old one if it exists
+      if (columnIndex === undefined) {
+        columnIndex = this._staticCellArray[staticIndex] && this._staticCellArray[staticIndex].props.columnIndex;
```

Comment:
so this is a typical case when old columns still present in our view's buffer are not present in the reducer buffer anymore
if we simply don't render them, it'll cause unmounts, so we just update them